deploy.sh

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -p jq -i bash
+
+# shellcheck shell=bash
+
+set -euo pipefail
+
+
+# For first deployment to bare server
+if [[ ${1:-} == '--prime' ]]; then
+    shift
+
+    BARE_SERVER=1
+else
+    BARE_SERVER=0
+fi
+
+get() {
+    jq -r ".$1"
+}
+
+ensure_set() {
+    if [[ "$(eval "echo \$$1")" == null ]]; then
+        echo "$2 is not set for profile $PROFILE of node $NODE"
+        exit 1
+    fi
+}
+
+deploy_profile() {
+    echo "=== Deploying profile $PROFILE of node $NODE ==="
+
+    ONLY_PROFILE="del(.path) | del(.activate)"
+    MERGED="$(jq "(. | del(.nodes) | del(.hostname) | $ONLY_PROFILE) + (.nodes.\"$NODE\" | del(.profiles) | $ONLY_PROFILE) + (.nodes.\"$NODE\".profiles.\"$PROFILE\" | del(.hostname))" <<< "$JSON")"
+    HOST="$(get hostname <<< "$MERGED")"
+    SSH_USER="$(get sshUser <<< "$MERGED")"
+    PROFILE_USER="$(get user <<< "$MERGED")"
+    CLOSURE="$(get path <<< "$MERGED")"
+    ACTIVATE="$(get activate <<< "$MERGED")"
+    EXTRA_SSH_OPTS="$(get sshOpts <<< "$MERGED")"
+
+    ensure_set HOST hostname
+    ensure_set CLOSURE path
+
+    if [[ "$EXTRA_SSH_OPTS" == null ]]; then
+        EXTRA_SSH_OPTS=""
+    fi
+
+    export NIX_SSHOPTS="${SSHOPTS:-} ${EXTRA_SSH_OPTS}"
+
+    SUDO=""
+
+    if [[ "$SSH_USER" == null ]]; then
+        if [[ "$PROFILE_USER" == null ]]; then
+            echo "neither user nor sshUser set for profile $PROFILE of node $NODE"
+            exit 1
+        fi
+        SSH_USER="$USER"
+    else
+        if [[ ! "$PROFILE_USER" == null ]] && [[ ! "$PROFILE_USER" == "$SSH_USER" ]]; then
+            SUDO="sudo -u $PROFILE_USER"
+        fi
+
+        if [[ "$USER" == null ]]; then
+            USER="$SSH_USER"
+        fi
+    fi
+
+
+    if [[ "$USER" == root ]]; then
+        PROFILE_PATH="/nix/var/nix/profiles/$PROFILE"
+    else
+        PROFILE_PATH="/nix/var/nix/profiles/per-user/$USER/$PROFILE"
+    fi
+
+    if [[ "$FLAKE_SUPPORT" == 1 ]]; then
+        nix build --no-link "$REPO#deploy.nodes.$NODE.profiles.$PROFILE.path"
```

Comment:
I think calling `nix build` is not needed here, the closure should be built when `nix eval` is called

Reply:
The closure isn't built by `nix eval`, I checked. Which is a good thing -- we don't want to build every single system configuration in a cluster every time we want to deploy just one.

Reply:
You're right, `nix eval` just puts the path